Sources/TuistDependencies/Carthage/Utils/CarthageGraphGenerator.swift

```diff
@@ -28,17 +28,30 @@ public final class CarthageGraphGenerator: CarthageGraphGenerating {
 
         let thirdPartyDependencies: [String: ThirdPartyDependency] = Dictionary(grouping: products, by: \.name)
             .compactMapValues { product in
-                guard let frameworkName = product.first?.container else { return nil }
-
-                let path = AbsolutePath("/")
-                    .appending(components: [
-                        Constants.tuistDirectoryName,
-                        Constants.DependenciesDirectory.name,
-                        Constants.DependenciesDirectory.carthageDirectoryName,
-                        frameworkName,
-                    ])
-
-                return .xcframework(path: path)
+                if let xcFrameworkName = product.first?.container {
+                    let path = AbsolutePath("/")
+                        .appending(components: [
+                            Constants.tuistDirectoryName,
+                            Constants.DependenciesDirectory.name,
+                            Constants.DependenciesDirectory.carthageDirectoryName,
+                            xcFrameworkName,
+                        ])
+
+                    return .xcframework(path: path)
+                } else if let frameworkName = product.first?.name {
+                    let path = AbsolutePath("/")
+                        .appending(components: [
+                            Constants.tuistDirectoryName,
+                            Constants.DependenciesDirectory.name,
+                            Constants.DependenciesDirectory.carthageDirectoryName,
+                            Constants.DependenciesDirectory.iOSDirectoryName,
```

Comment:
Shouldn't this depend on the platforms configured in the `Dependecies.swift`? this way we are only supporting iOS

Reply:
I think we should fix it in a separate PR and explained below